src/lambda/local/debugConfiguration.ts

```diff
@@ -33,3 +34,64 @@ export interface PythonDebugConfiguration extends DebugConfiguration {
         }
     ]
 }
+
+export interface DotNetCoreDebugConfiguration extends DebugConfiguration {
+    type: 'coreclr'
+    processId: string
+    pipeTransport: PipeTransport
+    windows: {
+        pipeTransport: PipeTransport
+    }
+    sourceFileMap: {
+        [key: string]: string
+    }
+}
+
+export interface PipeTransport {
+    pipeProgram: 'sh' | 'powershell'
+    pipeArgs: string[]
+    debuggerPath: '/tmp/lambci_debug_files/vsdbg',
+    pipeCwd: string
+}
+
+export function makeCoreCLRDebugConfiguration(
+    { codeUri, port }: {
+        port: number,
+        codeUri: string
+    }
+): DotNetCoreDebugConfiguration {
+    const pipeArgs = [
+        '-c',
+        `docker exec -i $(docker ps -q -f publish=${port}) \${debuggerCommand}`
+    ]
+    const debuggerPath = '/tmp/lambci_debug_files/vsdbg'
+
+    if (os.platform() === 'win32') {
+        // Coerce drive letter to uppercase. While Windows is case-insensitive, sourceFileMap is case-sensitive.
+        codeUri = codeUri.replace(/^\w\:/, match => match.toUpperCase())
+    }
+
+    return {
+        name: '.NET Core Docker Attach',
+        type: 'coreclr',
+        request: 'attach',
+        processId: '1',
+        pipeTransport: {
+            pipeProgram: 'sh',
+            pipeArgs,
+            debuggerPath,
+            pipeCwd: codeUri
+        },
+        windows: {
```

Comment:
Is isn't necessary since this configuration isn't saved anywhere (i.e. will only run on the current platform). But I wanted to minimize differences between the configuration that would appear in `launch.json` (see the [SAM CLI docs](https://github.com/awslabs/aws-sam-cli/blob/develop/docs/usage.md#debugging-applications)) and this configuration.